app/services/structure_updater.rb

```diff
@@ -101,14 +101,27 @@ def from_csv
 
     contains = files_by_fileset.keys.map do |sequence|
       # Find the existing fileset
-      existing_fileset = model.structural.contains[sequence.to_i - 1]
+      existing_fileset = file_set(sequence, files_by_fileset[sequence].first.label)
       attributes = fileset_attributes[sequence]
                    .merge(structural: { contains: files_by_fileset[sequence] })
       existing_fileset.new(**attributes)
     end
     Success(model.structural.new(contains:))
   end
 
+  def file_set(sequence, label)
+    model.structural.contains[sequence.to_i - 1].presence || new_file_set(label) # Cocina::Models::FileSet.new(files)
+  end
+
+  def new_file_set(label)
+    druid = model.externalIdentifier.delete_prefix('druid:')
+    fileset_id = "https://cocina.sul.stanford.edu/fileset/#{druid}-#{SecureRandom.uuid}"
+    Cocina::Models::FileSet.new(externalIdentifier: fileset_id,
```

Comment:
My initial question is: Are these defaults and generated ID sufficient and correct?